bin/linters.sh

```diff
@@ -19,82 +19,53 @@ while getopts :c: arg; do
-    fi
-
-    if [[ -z ${LAST_GOOD_GITSHA} ]] && [[ -n "${parent_branch}" ]]; then
-        LAST_GOOD_GITSHA="$(git log ${parent_branch}.. --pretty="%H"|tail -1)"
-        [[ ! -z ${LAST_GOOD_GITSHA} ]] && LAST_GOOD_GITSHA="${LAST_GOOD_GITSHA}^"
-    fi
-
-    # default: lint everything. This runs on the main build
-    if [[ -z ${PKGS} ]];then
-        PKGS="./adapter/... ./cmd/... ./example/... ./pkg/... ./tools/codegen/..."
-
-        # convert LAST_GOOD_GITSHA to list of packages.
-        if [[ ! -z ${LAST_GOOD_GITSHA} ]];then
-            echo "Using ${LAST_GOOD_GITSHA} to compare files to."
-            list=""
-            for fn in $(git diff --name-only ${LAST_GOOD_GITSHA}); do
-                # Always skip testdata and bin
-                case ${fn} in
-                    testdata/* | bin/*) : ;;
-                    *) { fd="${fn%/*}"; if [[ -d ${fd} ]]; then list="${list}\n${fd}"; fi; } ;;
-                esac
-            done
-            PKGS=$(echo -e "${list}" | sort | uniq)
-            echo -e "Running linters on: ${PKGS}\n"
-        else
-            echo "Running linters on all files."
-        fi
-    fi
+    NUM_CPU=$(getconf _NPROCESSORS_ONLN)
 
     # Note: WriteHeaderAndJson excluded because the interface is defined in a 3rd party library.
-    gometalinter.v1\
-        --concurrency=4\
+    gometalinter="docker run \
+      -v $(bazel info output_base):$(bazel info output_base) \
+      -v $(pwd):/go/src/istio.io/mixer \
+      -w /go/src/istio.io/mixer \
+      gcr.io/istio-testing/linter:bfcc1d6942136fd86eb6f1a6fb328de8398fbd80"
+    $gometalinter \
+        --concurrency=${NUM_CPU}\
         --enable-gc\
         --vendored-linters\
         --deadline=1200s --disable-all\
-        --enable=gosimple\
         --enable=aligncheck\
         --enable=deadcode\
         --enable=errcheck\
         --enable=gas\
         --enable=goconst\
         --enable=gofmt\
         --enable=goimports\
-        --enable=golint --min-confidence=0 --exclude=vendor --exclude=.pb.go --exclude=pkg/config/proto/combined.go --exclude=.*.gen.go --exclude="should have a package comment"\
+        --enable=golint --min-confidence=0\
+        --enable=gotype\
+        --exclude=vendor\
+        --exclude=.pb.go\
+        --exclude=pkg/config/proto/combined.go\
+        --exclude=.*.gen.go\
+        --exclude="should have a package comment"\
         --exclude=".*pkg/config/apiserver_test.go:.* method WriteHeaderAndJson should be WriteHeaderAndJSON"\
         --enable=ineffassign\
         --enable=interfacer\
         --enable=lll --line-length=160\
+        --enable=megacheck\
         --enable=misspell\
-        --enable=staticcheck\
         --enable=structcheck\
         --enable=unconvert\
         --enable=unparam\
-        --enable=unused\
         --enable=varcheck\
         --enable=vet\
         --enable=vetshadow\
         --skip=testdata\
         --skip=vendor\
         --vendor\
-        $PKGS
+        ./...
```

Comment:
We do want to maitain he option of running it on specific packages.

Reply:
it only takes 30s to run it on all of codebase. I can use PKGS as an imported env variable.

Reply:
It takes _way_ more than 30 seconds when running on a Mac. Many many minutes...

Reply:
This patch should make the linter extremely faster than the one on master branch. As I reported in #1155, gometalinter.v1 is 14x slower than the recent version, because of target file selection.

It took ~1min on my environment _including_ fetching docker images. Subsequent calls took ~30 secs.

Reply:
This works great for prow and linux machines, but it's not so great on mac. I tried it on my macbook and docker there is rather slow. I'm thinking of dropping to using native gometalinter on mac? Would that work for you guys?